### PR TITLE
Invite People: limit Message to 500 characters

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #15356 

This limits the custom message to 500 characters. 

Tech side note: The `NSLocalizedString`s are kind of scattered throughout, and some are duplicated/very similar. In a follow-up PR, I'll look into extracting them into a `Constants` or something. Something less repetitive anyway. 😄 

To test:

Character limiting:
- Go to Site > People.
- Tap '+' in the nav bar.
- Tap on the `Custom message...` field.
- Type an opus exceeding 500 characters. Or, in `InvitePersonViewController`, change this value to something manageable: `private let messageCharacterLimit = 500`
- Verify once the limit is reached, no more characters are entered.

_Note_: there is an outstanding issue (#10795) where the entire message is not displayed in the invite Message view.

Footer text:
- Verify on both views (the Invite view and the Message entry view) the footer text now references the character limit.

| invite | message |
|--------|-------|

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
Hey @mattmiklic . Since iOS doesn't have the character count on the message field (`xxx characters remaining`), I took the liberty of adding the restriction in the footer text. Please see the screenshots in the description. Let me know what you think. Thanks!

**Review by mattmiklic — APPROVED** [review]
This looks good to me! Thanks @ScoutHarris.

File: WordPress/Classes/ViewRelated/Tools/SettingsMultiTextViewController.m [hunk 1]
```diff
@@ -160,4 +160,26 @@ - (void)adjustCellSize
...
+    // If character length is unrestricted, allow all text.
+    if (self.maxCharacterCount <= 0) {
+        return YES;
+    }
+    
+    NSString *newText = [textView.text stringByReplacingCharactersInRange: range withString: text];
+    
+    // If the entire new text will fit, allow it.
+    if (newText.length <= self.maxCharacterCount)
+    {
```
Comment (mindgraffiti) [inline WordPress/Classes/ViewRelated/Tools/SettingsMultiTextViewController.m]: nit
```suggestion
    if (newText.length <= self.maxCharacterCount) {
```

**Review by mindgraffiti — APPROVED** [review]
✅  code
✅  UI tested

:shipit:

File: WordPress/Classes/ViewRelated/Tools/SettingsMultiTextViewController.m (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Tools/SettingsMultiTextViewController.m]: Thanks @mindgraffiti . I had to fix another one just I just did another commit.

**Pull request merged**